speechbrain/lobes/models/discrete_tokens/dac.py

```diff
@@ -0,0 +1,1108 @@
+        strict: bool = False,
+        load_pretrained: bool = False,
+    ):
+        """ Initializes DAC
+
+        Parameters
+        ----------
+        encoder_dim : int, optional, by default 64
+        encoder_rates : List[int], optional, by default [2, 4, 8, 8]
+        latent_dim : int, optional, by default None
+        decoder_dim : int, optional, by default 1536
+        decoder_rates : List[int], optional, by default [8, 8, 4, 2]
+        n_codebooks : int, optional, by default 9
+        codebook_size : int, optional, by default 1024
+        codebook_dim : Union[int, list], optional, by default 8
+        quantizer_dropout : bool, optional, by default False
+        sample_rate : int, optional, by default 44100
+        model_type : str, optional, by default "44khz"
+        model_bitrate : str, optional, by default "8kbps"
+        tag : str, optional, by default "latest"
+        load_path : str, optional, by default None
+        strict : bool, optional, by default False
+        load_pretrained : bool, optional
+             If True, then a pretrained model is loaded, by default False
+        """
+        super().__init__()
+
+        self.encoder_dim = encoder_dim
+        self.encoder_rates = encoder_rates
+        self.decoder_dim = decoder_dim
+        self.decoder_rates = decoder_rates
+        self.sample_rate = sample_rate
+        self.n_codebooks = n_codebooks
+        self.codebook_size = codebook_size
+        self.codebook_dim = codebook_dim
+        self.latent_dim = latent_dim
+        self.quantizer_dropout = quantizer_dropout
+
+        if load_pretrained:
+            if not load_path:
+                load_path = download(
+                    model_type=model_type, model_bitrate=model_bitrate, tag=tag
+                )
+                print(f"Obtained load path as: {load_path}")
+            model_dict = torch.load(load_path, "cpu")
+            metadata = model_dict["metadata"]
+            for key, value in metadata["kwargs"].items():
+                setattr(self, key, value)
+
+        self.hop_length = np.prod(self.encoder_rates)
+        if self.latent_dim is None:
+            self.latent_dim = self.encoder_dim * (2 ** len(self.encoder_rates))
+        self.encoder = Encoder(
+            self.encoder_dim, self.encoder_rates, self.latent_dim
+        )
+        self.quantizer = ResidualVectorQuantize(
+            input_dim=self.latent_dim,
+            n_codebooks=self.n_codebooks,
+            codebook_size=self.codebook_size,
+            codebook_dim=self.codebook_dim,
+            quantizer_dropout=self.quantizer_dropout,
+        )
+        self.decoder = Decoder(
+            self.latent_dim, self.decoder_dim, self.decoder_rates,
+        )
+        self.apply(init_weights)
+
+        if load_pretrained:
+            self.load_state_dict(model_dict["state_dict"], strict=strict)
+            self.metadata = metadata
+
+    def preprocess(self, audio_data: torch.Tensor, sample_rate: int):
```

Comment:
Most of the time in speech brain, the resampling and the padding have been done in the dataio_prepare of each recipe..So maybe this function is not necessary but the assertion for the sampling rate is good to be done here and also it would be good that you mention the required SR  in the docstring.  also, it might be better to do this check inside the forward fiction.

Also, you need to set the default value for sample_rate to None.. otherwise dac.preprocess(audio_data) yields " missing 1 required positional argument: 'sample_rate"